WorldModel/WorldModel/Core/CoreFunctions.aslx

```diff
@@ -568,10 +568,14 @@
     return (l)
   </function>
   
-  <!-- Added by KV -->
-
   <function name="AllRooms" type="objectlist">
-    return (FilterByAttribute(AllObjects(),"isroom",true))
+    l = NewObjectList()
+    foreach (o, AllObjects()) {

```

Comment:
This `foreach` is the same thing (or *should* be the same thing) as `FilterByAttribute(AllObjects(),"isroom",true)`

https://github.com/textadventures/quest/blob/8625f9b071acee23efb18514ae68b398de722f29/WorldModel/WorldModel/Core/CoreFunctions.aslx#L532-L542

---
Did it not return the same list the way it was? (Just curious.)

Reply:
no, I have to add the "is object and room" which does not have the attribute isroom

Reply:
So... Why does mine work? I don't understand that.

The only thing I see different is I added `isroom` at the end of the `defaultobject` type's attributes, and you have yours at the very beginning.

YOURS:
![image](https://github.com/user-attachments/assets/3a511e90-d266-4aad-97aa-64df5c6194d2)

---

MINE:
![image](https://github.com/user-attachments/assets/ed67ec83-dd0c-48ca-b185-93b7ba8e487f)

---
And our `default_object` and `default_room` bits are inverted:

YOURS:
![image](https://github.com/user-attachments/assets/60e83257-9673-46a3-9d19-d0ceb1844957)

---

MINE:
![image](https://github.com/user-attachments/assets/c164c60e-6f23-411b-9486-4b8ca91e3604)

---
I didn't do any of that on purpose, either.